examples/mp-metrics-timed/src/test/java/WeatherServiceTest.java

```diff
@@ -0,0 +1,151 @@
+        assertTrue(metric.contains("# TYPE application:weather_day_status_seconds timer"));
+        assertTrue(metric.contains("application:weather_day_status_seconds{quantile=\"0.95\"}"));
+        assertTrue(metric.contains("# TYPE application:weather_day_status_seconds timer"));
+        assertTrue(metric.contains("application:weather_day_status_seconds{quantile=\"0.98\"}"));
+        assertTrue(metric.contains("# TYPE application:weather_day_status_seconds timer"));
+        assertTrue(metric.contains("application:weather_day_status_seconds{quantile=\"0.99\"}"));
+        assertTrue(metric.contains("# TYPE application:weather_day_status_seconds timer"));
+        assertTrue(metric.contains("application:weather_day_status_seconds{quantile=\"0.999\"}"));
+    }
+
+    private void assertJsonFormat(final String metricPath) {
+        final String metric = WebClient.create(base.toExternalForm())
+                .path(metricPath)
+                .accept(MediaType.APPLICATION_JSON)
+                .get(String.class);
+
+        assertTrue(metric.contains("count"));
+        assertTrue(metric.contains("meanRate"));
+        assertTrue(metric.contains("fifteenMinRate"));
+        assertTrue(metric.contains("fiveMinRate"));
+        assertTrue(metric.contains("oneMinRate"));
+        assertTrue(metric.contains("min"));
+        assertTrue(metric.contains("max"));
+        assertTrue(metric.contains("mean"));
+        assertTrue(metric.contains("stddev"));
+        assertTrue(metric.contains("p50"));
+        assertTrue(metric.contains("p75"));
+        assertTrue(metric.contains("p95"));
+        assertTrue(metric.contains("p98"));
+        assertTrue(metric.contains("p99"));
+        assertTrue(metric.contains("p999"));
+
+    }
+
+    @Test
+    public void testTimedMetricMetadata() {
+        final Response response = WebClient.create(base.toExternalForm())
+                .path("/metrics/application/weather_day_status")
+                .accept(MediaType.APPLICATION_JSON)
+                .options();
+        final String metaData = response.readEntity(String.class);
+        JsonObject metadataJson = Json.createReader(new StringReader(metaData)).readObject();
+
+        final String expected = "{\n" +
+                "  \"weather_day_status\": {\n" +
+                "    \"description\": \"This metric shows the weather status of the day.\",\n" +
+                "    \"displayName\": \"Weather Day Status\",\n" +
+                "    \"name\": \"weather_day_status\",\n" +
+                "    \"reusable\": false,\n" +
+                "    \"tags\": \"\",\n" +
+                "    \"type\": \"timer\",\n" +
+                "    \"typeRaw\": \"TIMER\",\n" +
+                "    \"unit\": \"nanoseconds\"\n" +
+                "  }\n" +
+                "}";
+
+        JsonObject expectedJson = Json.createReader(new StringReader(expected)).readObject();
+        assertEquals(expectedJson, metadataJson);
```

Comment:
this is not stable so either assert using keys (keys1.size() == keys2.size() && for all key value ==) or using another way to write the assertEquals

Reply:
Done